### PR TITLE
Migrate tests to modern JUnit Jupiter

### PR DESCRIPTION
**Pull request by mureinik**
The project currently uses the outdated JUnit 4.13.2. In order to make it easier to write tests and easier for future contributors to easily start working with the project, this path migrates the test suite to the modern JUnit Jupiter.

Changes:
- The `junit:junit:4.13.2` dependency was replaced with `org.junit.jupiter:junit-jupiter:5.8.1`.
- The `org.junit.jupiter.api.Test` annotation was used as a drop-in replacement for `org.junit.Test`.
- The static imports from `org.junit.jupiter.api.Assertions` were used as drop-in replacements for the static imports from `org.junit.Assert`.
- `LineEndingEdgeCasesTest` was rewritten to use JUnit Jupiter's `org.junit.jupiter.api.io.TempDir` annotation instead of JUnit 4's `org.junit.rules.TemporaryFolder`.

### CODE REVIEW

File: src/test/java/net/revelc/code/impsort/LineEndingEdgeCasesTest.java [hunk 1]
```diff
@@ -40,16 +39,15 @@
   private static Grouper eclipseDefaults =
       new Grouper("java.,javax.,org.,com.", "", false, false, true);
 
-  @Rule
-  public TemporaryFolder folder =
-      new TemporaryFolder(new File(System.getProperty("user.dir"), "target"));
+  @TempDir
+  public File folder;
```
Comment (ctubbsii) [inline src/test/java/net/revelc/code/impsort/LineEndingEdgeCasesTest.java:43]: I'd strongly prefer our temporary files be located under the Maven build directory, `target/`. This could be done with `TemporaryFolder`, but looks like `TempDir` is limited. See junit-team/junit5#2088

This is probably an acceptable change for this project, but if there's something that can be done to keep it in `target/` instead of `/tmp`, I'd be relieved.

**Review by ctubbsii — APPROVED** [review]
Thanks for the contribution, @mureinik !

I would have preferred we keep the temp directory under `target/`, but this change is okay as-is.

File: src/test/java/net/revelc/code/impsort/LineEndingEdgeCasesTest.java (hunk 1 above)
Comment (mureinik) [reply src/test/java/net/revelc/code/impsort/LineEndingEdgeCasesTest.java:43]: I couldn't find a way to do it, so I thought I'd post a PR without it and see what the maintainers think, and possibly start a discussion. But I see you merged it already :-)

One thing that might be possible is to override the `java.io.tmpdir` for this test and force `@TempDir` to create it under `target/`. Let me play around with it a bit and see if this is a viable option.

Comment (mureinik) [reply src/test/java/net/revelc/code/impsort/LineEndingEdgeCasesTest.java:43]: @ctubbsii Following up - The JVM reads `java.io.tmpdir` when it initializes the `jdk.internal.util.StaticProperty` class, and `java.nio.file.TempFileHelper` then uses that. 
TL;DR - while it's not hard to overwrite `java.io.tmpdir` during runtime, that won't affect where JUnit's `@TempDir` creates temporary files, so that idea is a bust.

Another direction is to abandon that annotation and have explicit code in a `@BeforeEach` method to create a temp dir under `target` and then tear it down in an `@AfterEach` method. I'm not sure it's worth the need to maintain this code later on, but if you think it is, I'll gladly send a followup patch. Let me know what you think.

Comment (ctubbsii) [reply src/test/java/net/revelc/code/impsort/LineEndingEdgeCasesTest.java:43]: I think it's possible to set `java.io.tmpdir` in the `maven-surefire-plugin` so it will affect the test that way, but I have not yet verified that.

Comment (mureinik) [reply src/test/java/net/revelc/code/impsort/LineEndingEdgeCasesTest.java:43]: Good point! I didn't think of doing that.
Note that it won't help for the usecase you manually run the test from your IDE, but I guess that's less of a concern.

Here's my attempt at it, let me know if that's what you meant: #52
